src/main/java/org/opensearch/security/tools/democonfig/Installer.java

```diff
@@ -0,0 +1,377 @@
+                }
+
+                if (!cluster_mode) {
+                    System.out.println("Cluster mode requires additional setup of:");
+                    System.out.println("  - Virtual memory (vm.max_map_count)\n");
+                    cluster_mode = confirmAction(scanner, "Enable cluster mode?");
+                }
+            }
+        } else {
+            initsecurity = true;
+            cluster_mode = true;
+        }
+    }
+
+    /**
+     * Helper method to scan user inputs.
+     * @param scanner object to be used for scanning user input
+     * @param message prompt question
+     * @return true or false based on user input
+     */
+    static boolean confirmAction(Scanner scanner, String message) {
+        System.out.print(message + " [y/N] ");
+        String response = scanner.nextLine();
+        return response.equalsIgnoreCase("yes") || response.equalsIgnoreCase("y");
+    }
+
+    /**
+     * Initialize all class level variables required
+     */
+    static void initializeVariables() {
+        setBaseDir();
+        setOpenSearchVariables();
+        setSecurityVariables();
+    }
+
+    /**
+     * Sets the base directory to be used by the script
+     */
+    static void setBaseDir() {
+        File baseDirFile = new File(SCRIPT_DIR).getParentFile().getParentFile().getParentFile();
+        BASE_DIR = baseDirFile != null ? baseDirFile.getAbsolutePath() : null;
+
+        if (BASE_DIR == null || !new File(BASE_DIR).isDirectory()) {
+            System.out.println("DEBUG: basedir does not exist");
+            System.exit(-1);
+        }
+
+        BASE_DIR += File.separator;
+    }
+
+    /**
+     * Sets the variables for items at OpenSearch level
+     */
+    static void setOpenSearchVariables() {
+        OPENSEARCH_CONF_FILE = BASE_DIR + "config" + File.separator + "opensearch.yml";
+        OPENSEARCH_BIN_DIR = BASE_DIR + "bin" + File.separator;
+        OPENSEARCH_PLUGINS_DIR = BASE_DIR + "plugins" + File.separator;
+        String OPENSEARCH_MODULES_DIR = BASE_DIR + "modules" + File.separator;
+        OPENSEARCH_LIB_PATH = BASE_DIR + "lib" + File.separator;
+        OPENSEARCH_INSTALL_TYPE = determineInstallType();
+
+        if (!(new File(OPENSEARCH_CONF_FILE).exists())) {
+            System.out.println("Unable to determine OpenSearch config directory. Quit.");
+            System.exit(-1);
+        }
+
+        if (!(new File(OPENSEARCH_BIN_DIR).exists())) {
+            System.out.println("Unable to determine OpenSearch bin directory. Quit.");
+            System.exit(-1);
+        }
+
+        if (!(new File(OPENSEARCH_PLUGINS_DIR).exists())) {
+            System.out.println("Unable to determine OpenSearch plugins directory. Quit.");
+            System.exit(-1);
+        }
+
+        if (!(new File(OPENSEARCH_MODULES_DIR).exists())) {
+            System.out.println("Unable to determine OpenSearch modules directory. Quit.");
+            // System.exit(-1);
```

Comment:
This looks like it should not be commented out.

Reply:
I translated the script as is. But upon digging further I see that this constant is never used after assignment. I'll remove it completely